### PR TITLE
Enables password SSH auth, fixes #1867

### LINKED ISSUE #1867: MySQL with SSH tunnel results to fatal error
In my case the MySQL connection information provided [here](https://github.com/Varying-Vagrant-Vagrants/VVV/wiki/Connecting-to-MySQL-MariaDB#ssh-tunnel) are no longer relevant.

I'm using HeidiSQL, which fails to connect to my VMs mariadb with an error message:

```
PLink exited unexpected. Command line was:
"c:\Program Files\PuTTY\plink.exe" -ssh vagrant@vvv.test -pw "vagrant" -N -L 3307:127.0.0.1:3306
```

When running this command from the command line, I get

```
"c:\Program Files\PuTTY\plink.exe" -ssh vagrant@vvv.test -pw "vagrant" -N -L 3307:127.0.0.1:3306
Using username "vagrant".
FATAL ERROR: Disconnected: No supported authentication methods available (server sent: publickey)
```

Note that the exact same configuration was working properly with my previous VM.

### PR DESCRIPTION
**Pull request by tomjn**
## Summary:

This attempts to fix #1867

To test, run:

```sh
git checkout origin fix/dbssh
vagrant provision
```

Then attempt to connect to the database from an SQL client on the host

## Checks

 - [x] I've tested this PR with Vagrant **v2.2.4** and VirtualBox **v6.0.8** on **MacOS**
 - [x] This PR is for the `develop` branch not the `master` branch
 - [x] I've updated the changelog
 - [x] This PR is complete and ready for review

### CODE REVIEW

**Comment by tomjn** [comment]
Of note, it might require a `vagrant halt` and a `vagrant up`, I'm not sure if the change takes immediate effect or not

**Comment by tomjn** [comment]
@Mte90 does the change take effect immediatley or do we need to restart the SSH agent?

**Comment by Mte90** [comment]
Require a restart of the sshd daemon so better a up and halt.

**Comment by tomjn** [comment]
@Mte90 then we should probably do that in the script so that it works on the first provision

**Comment by tomjn** [comment]
I've made the change, but I'm concerned it might sever any active connections, halting provisioning in the process

**Comment by Mte90** [comment]
well during provision is better to no do anything with the machine to avoid issues so I think that is a best practice that in this case confirm that case 😁

File: provision/provision.sh [hunk 1]
```diff
@@ -282,7 +282,10 @@ profile_setup() {
   fi
 
   echo " * Copying /srv/config/ssh_known_hosts to /etc/ssh/ssh_known_hosts"
- cp -f /srv/config/ssh_known_hosts /etc/ssh/ssh_known_hosts
+  cp -f /srv/config/ssh_known_hosts /etc/ssh/ssh_known_hosts
+  echo " * Enabling SSH Password Authentication for database applications"
+  echo "PasswordAuthentication yes" >> /etc/ssh/sshd_config
```
Comment (tomjn) [inline provision/provision.sh]: oh dear, I realise now this file is going to get very large with the same line repeated over and over again

**Comment by tomjn** [comment]
So this works for me, but the appending to the config needs changing so it only happens if it's needed

File: provision/provision.sh (hunk 1 above)
Comment (Mte90) [reply provision/provision.sh]: maybe with `cat /etc/ssh/sshd_config | grep 'PasswordAuthentication yes'``we can do a check if it is already on it

**Comment by Mte90** [comment]
```
if [[ ! ((cat /etc/ssh/sshd_config | grep 'PasswordAuthentication yes')); then
    echo " * Enabling SSH Password Authentication for database applications"
    echo "PasswordAuthentication yes" >> /etc/ssh/sshd_config
    systemctl restart ssh
  fi
```
should fix the behaviour

**Comment by tomjn** [comment]
Solved by copying `sshd_config` into place

**Comment by Mte90** [comment]
sounds better

**Comment by kmgalanakis** [comment]
Just confirming the fix. Thanks ❤️

**Pull request merged**